src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IWorkspaceProjectContext.cs

```diff
@@ -76,6 +77,7 @@ internal interface IWorkspaceProjectContext : IDisposable
 
         void StartBatch();
         void EndBatch();
+        ValueTask EndBatchAsync();
```

Comment:
1. do we want both this and the Sync version?
2. do we need StartBatchAsync?

Reply:
We don't want the sync version but yes forgot to make it obsolete.

For StartBatchAsync, at this point I don't think so. The locking design here for the workspace is each project has it's own lock, and then there's also the big workspace lock when we're actually making changes. The intent with a batch is each of the operations within the batch don't require the big workspace lock until you actually apply it -- hence the EndBatchAsync is being made async here to avoid contention on the big global lock. StartBatch only is taking the project lock, which is unlikely to have any contention unless the project system is updating a single project from multiple threads at once.

tl;dr: the idea of batching is everything is cheap until the EndBatch, so we really shouldn't need it.